### PR TITLE
Customizable config file with ability to use local aws-proxy to access OpenSearch

### PR DESCRIPTION
**Pull request by johnnyon-amzn**
### Description
Set environment variables based on local config file and access OpenSearch with proxy credentials
### Issues Resolved
https://github.com/opensearch-project/dashboards-desktop/issues/11

### CODE REVIEW

**Comment by johnnyon-amzn** [comment]
Squashing for readability. Electron app with customizable config and ability to connect to aws proxy

**Comment by seraphjiang** [comment]
Great progress,
a few comments in general

1. move the code to src folder instead of put on project root folder
2. could we add a few screen shot about what look UI looks like so far

File: renderer.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+function getMapping() {
+    return [{id: "awsAccessKeyId", key: "AWS_ACCESS_KEY_ID"},
+            {id: "awsSecretAccessKey", key: "AWS_SECRET_ACCESS_KEY"},
+            {id: "endpoint", key: "ENDPOINT"},
+            {id: "osdPath", key: "osdPath"}
```
Comment (zhongnansu) [inline renderer.js:5]: Do we need extra logic to validate that the path provided is actually an OSD path? What if user put some other random links or other aws resources links

**Review by zhongnansu — COMMENTED** [review]
Shouldn't we add license headers to all files?

File: renderer.js (hunk 1 above)
Comment (johnnyon-amzn) [reply renderer.js:5]: I will look into adding validation. I think the simplest would be to try to run the curl command and just output the stderr.

File: README.md [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+- [Copyright](#copyright)
 
-Be sure to:
+## Welcome
 
-* Change the title in this README
-* Edit your repository description on GitHub
+Dashboards Desktop offers desktop experiene to OpenSearch Dashboards Users. OpenSearch Dashboards is an open source search and analytics visualization application. 
+
+Dashboards Desktop provides end user *flexibility* and *extensibility* to install any Dashboards pluings they want. Dashboards Desktop could store data into local storage or cloud drive. User has full control of the data they have.
```
[the remark this answers is not in the document]
Comment (johnnyon-amzn) [reply README.md]: Update in commit 8fb16b4

File: README.md [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+
+Dashboards Desktop could be downloaded from internet. Not only deliver stable version to customer in monthly/quarterly release cadence. It will deliver experimental versions.
+
+Deskboards has build-in data source selector which allow user to connect to any OpenSearch endpoint they want with pre-configured credential or access key.
```
[the remark this answers is not in the document]
Comment (johnnyon-amzn) [reply README.md]: Update in commit 8fb16b4

File: config.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"AWS_ACCESS_KEY_ID":"","AWS_SECRET_ACCESS_KEY":"","ENDPOINT":"","osdPath":""}
```
[the remark this answers is not in the document]
Comment (johnnyon-amzn) [reply config.json]: Update in commit 8fb16b4

**Comment by johnnyon-amzn** [comment]
> Great progress, a few comments in general
> 
> 1. move the code to src folder instead of put on project root folder
> 2. could we add a few screen shot about what look UI looks like so far

I migrated all code to the src folder, and changed the paths in commit 0924bed, 8fb16b4
I also added a screenshot for the initial page in commit 53c9349. I will add more screenshots for the next commits with opensearch dashboards connecting.

**Review by seraphjiang — APPROVED** [review]
Consider move css folder to src as well

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.DS_Store
+config.json
```
Comment (Flyingliuhub) [inline .gitignore:3]: Is there any reason that we need to ignore config.json file? Thanks

File: package.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  "name": "dashboards-desktop",
+  "version": "1.0.0",
+  "description": "<img src=\"https://opensearch.org/assets/brand/SVG/Logo/opensearch_dashboards_logo_darkmode.svg\" height=\"64px\"/>",
+  "main": "src/main.js",
+  "scripts": {
+    "start": "electron ."
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/johnnyon-amzn/dashboards-desktop-poc.git"
```
Comment (Flyingliuhub) [inline package.json:11]: Is it possible not have personal information "johnnyon-amzn"? I thought that should point to "opensearch-project/dashboards-desktop", the same as other url.

**Review by Flyingliuhub — APPROVED** [review]

File: .gitignore (hunk 5 above)
Comment (johnnyon-amzn) [reply .gitignore:3]: Config file has sensitive information like AWS_SECRET_KEY. I created a template file as a placeholder

File: package.json (hunk 6 above)
Comment (johnnyon-amzn) [reply package.json]: I updated in commit a1eb3eb to reference main github page

**Review by zhongnansu — APPROVED** [review]
lgtm

**Pull request merged**